controllers/global_search_setup.go

```diff
@@ -0,0 +1,558 @@
+						map[string]interface{}{
+							"apiVersion": "route.openshift.io/v1",
+							"kind":       "Route",
+							"metadata": map[string]interface{}{
+								"name":      "search-global-hub",
+								"namespace": "open-cluster-management",
+								"labels": map[string]interface{}{
+									"app": "search",
+								},
+							},
+							"spec": map[string]interface{}{
+								"port": map[string]interface{}{
+									"targetPort": "search-api",
+								},
+								"tls": map[string]interface{}{
+									"termination": "passthrough",
+								},
+								"to": map[string]interface{}{
+									"kind": "Service",
+									"name": "search-search-api",
+									// "weight": 100,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	_, err := r.DynamicClient.Resource(manifestWorkGvr).Namespace(cluster).
+		Create(ctx, manifestWork, metav1.CreateOptions{})
+
+	if err != nil && errors.IsAlreadyExists(err) {
+		log.V(5).Info("Found existing ManifestWork search-global-config for", "cluster", cluster)
+		return nil
+	}
+	return err
+}
+
+// Logic to disable Global Search.
+//  1. Disable global search feature in the console.
+//  2. Disable federated search feature in the search-api deployment.
+//  3. Delete configuration resources for each Managed Hub.
+func (r *SearchReconciler) disableGlobalSearch(ctx context.Context, instance *searchv1alpha1.Search) error {
+	// 1. Disable global search feature in the console.
+	// 1a. Remove the globalSearchFeatureFlag key to configmap console-mce-config in multicluster-engine namespace.
+	err := r.updateConsoleConfig(ctx, false, "multicluster-engine", "console-mce-config")
+	if err != nil {
+		log.Error(err, "Failed to remove the globalSearchFeatureFlag in configmap console-mce-config.")
+	}
+
+	// 1b. Remove the globalSearchFeatureFlag key to configmap console-config in open-cluster-management namespace.
+	err = r.updateConsoleConfig(ctx, false, instance.GetNamespace(), "console-config")
+	if err != nil {
+		log.Error(err, "Failed to remove the globalSearchFeatureFlag in configmap console-config.")
+	}
+
+	// 2. Disable federated search feature in the search-api.
+	// oc patch search search-v2-operator -n open-cluster-management --type='merge'
+	//    -p '{"spec":{"deployments":{"queryapi":{"envVar":[{"name":"FEATURE_FEDERATED_SEARCH", "value":"false"}]}}}}'
+	err = r.updateSearchApiDeployment(ctx, false, instance)
+	if err != nil {
+		log.Error(err, "Failed to remove the federated global search feature flag from search-api deployment.")
+	}
+
+	// 3. Delete configuration resources for each Managed Hub.
+	// MANAGED_HUBS=($(oc get managedcluster -o json | jq -r '.items[] | select(.status.clusterClaims[] |
+	//    .name == "hub.open-cluster-management.io" and .value != "NotInstalled") | .metadata.name'))
+	clusterList, err := r.DynamicClient.Resource(managedClusterResourceGvr).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Error(err, "Failed to list ManagedClusters.")
+	}
```

Comment:
Put this in else or check if  clusterList is not nil and items are present? Wondering if it will go into 'nilPointer' error otherwise.

Reply:
FYI @jlpadilla

Reply:
Resolved with latest commit. Also updated the delete logic to simplify and reduce api requests.